src/WebJobs.Extensions.DurableTask/Listener/WrappedFunctionResult.cs

```diff
@@ -17,9 +17,11 @@ internal class WrappedFunctionResult
 
         internal enum FunctionResultStatus
         {
-            Success = 0,
-            UserCodeError = 1,
-            FunctionsRuntimeError = 2,
+            Success = 0, // the function executed successfully.
+            UserCodeError = 1, // the user code had an unhandled exception
+            FunctionsRuntimeError = 2, // the runtime could not execute the user code for some reason; considered transient and retried
+            FunctionTimeoutError = 3, // execution timed out; treated as a user error
+            FunctionsHostStoppingError = 4, // host was shutting down; treated as a functions runtime error
```

Comment:
Why distinguish `FunctionsHostStoppingError` from the existing `FunctionsRuntimeError`? It looks to me like you're checking for both of these together each time.

Reply:
You are correct, they are not currently handled any differently. 

I just found it somewhat helpful to break this down so the code is explaining itself a bit more. It is conceivable we might use this information somewhere.